src/SocialAuthUserManager.php

```diff
@@ -156,20 +167,22 @@ public function setSessionKeysToNullify(array $session_keys) {
    *   The social implementer identifier.
    * @param string $provider_user_id
    *   The unique id returned by the user.
+   * @param JSON $data
+   *   The additional user_data to be stored in database.
    * @param string|bool $picture_url
    *   The user's picture.
    *
    * @return \Symfony\Component\HttpFoundation\RedirectResponse
    *   A redirect response.
    */
-  public function authenticateUser($name, $email, $pluginId, $provider_user_id, $picture_url = FALSE) {
+  public function authenticateUser($name, $email, $pluginId, $provider_user_id, $data, $picture_url = FALSE) {
```

Comment:
To make it backward compatible, I would put $data with a default value at the end